practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/converter/FhirPractitionerConverter.java

```diff
@@ -0,0 +1,41 @@
+package uk.ac.ox.ndph.mts.practitioner_service.converter;
+
+import org.hl7.fhir.r4.model.HumanName;
+import org.springframework.stereotype.Component;
+import uk.ac.ox.ndph.mts.practitioner_service.model.Practitioner;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Implement an EntityConverter for Practitioner
+ */
+@Component
+public class FhirPractitionerConverter implements EntityConverter<org.hl7.fhir.r4.model.Practitioner, Practitioner> {
+    /**
+     * Convert an hl7 Practitioner to a model Practitioner.
+     *
+     * @param input the practitioner to convert.
+     * @return uk.ac.ox.ndph.mts.practioner_service.model.Practitioner
+     */
+    public Practitioner convert(org.hl7.fhir.r4.model.Practitioner input) {
+        Optional<HumanName> practitionerName = input.getName().stream().findFirst();
```

Comment:
Using Optional#isPresent (in non-library code) is usually a code smell. In this case you could use Optional#map to do the conversion if present, then Optional#orElse or Optional#orElseGet instead. Or just not use an Optional here and check for an empty getName list instead.

Reply:
I changed it a bit. Let me know if it works.